tests/api2/test_475_syslog.py

```diff
@@ -0,0 +1,92 @@
+import pytest
+from auto_config import dev_test, ip, password, user
+from middlewared.test.integration.utils import ssh
+from pytest_dependency import depends
+from time import sleep
+
+pytestmark = pytest.mark.skipif(dev_test, reason='Skipping for test development testing')
+
+
+def do_syslog(ident, message, facility='syslog.LOG_USER', priority='syslog.LOG_INFO'):
+    """
+    This generates a syslog message on the TrueNAS server we're currently testing.
+    We don't need to override IP addr or creds because we are not a syslog target.
+    """
+    cmd = 'python3 -c "import syslog;'
+    cmd += f'syslog.openlog(ident=\\\"{ident}\\\", facility={facility});'
+    cmd += f'syslog.syslog({priority},\\\"{message}\\\");syslog.closelog()"'
+    ssh(cmd)
+
+
+def check_syslog(log_path, message, target_ip=ip, target_user=user, target_passwd=password)):
+    """
+    Common function to check whether a particular message exists in a log file.
+    This will be used to check local and remote syslog servers.
+
+    Current implementation performs simple grep through the log file, and so
+    onus is on test developer to not under-specify `message` in order to avoid
+    false positives.
+    """
+    cmd = f'grep -R "{message}" {log_path}'
+    ssh(cmd, user=target_user, password=target_passwd, ip=target_ip) 
```

Comment:
You don't need to pass this

Reply:
We should have remote syslog server available in CI pipeline soon. Once this is in place the tests will be expanded to validate that messages are appearing properly on remote server.